File: base/src/main/java/com/zhangteng/base/bean/GroupInfo.java
package com.zhangteng.base.bean;

/**
 * Created by swing on 2018/4/12.
 */
public class GroupInfo {
    /**
     * 按26个英文字母分组
     * 存储26个组的总数
     * 多的一个备用
     */
    public static int[] totals = new int[27];
    private int groupNum;
    private String title;
    private int position;
    private int total;

    public GroupInfo() {
    }

    public GroupInfo(int groupNum, String title, int position, int total) {
        this.groupNum = groupNum;
        this.title = title;
        this.position = position;
        this.total = total;
    }

    public GroupInfo(Long groupNum, String title, Long position, Long total) {
        this.groupNum = groupNum.intValue();
        this.title = title;
        this.position = position.intValue();
        this.total = total.intValue();
    }

    public static void initTotals() {
        for (int i = 0; i < totals.length; i++) {
            totals[i] = 0;
        }
    }

    public boolean isFirst() {
        return position == 0;
    }

    public boolean isLast() {
        return position == total - 1;
    }

    public int getGroupNum() {
        return groupNum;
    }

    public void setGroupNum(int groupNum) {
        this.groupNum = groupNum;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public int getPosition() {
        return position;
    }

    public void setPosition(int position) {
        this.position = position;
    }

    public int getTotal() {
        return total;
    }

    public void setTotal(int total) {
        this.total = total;
    }
}

File: base/src/main/java/com/zhangteng/base/bean/GroupInfo.kt
package com.zhangteng.base.bean

/**
 * Created by swing on 2018/4/12.
 */
open class GroupInfo {
    private var groupNum = 0
    private var title: String? = null
    private var position = 0
    private var total = 0

    constructor()
    constructor(groupNum: Int, title: String?, position: Int, total: Int) {
        this.groupNum = groupNum
        this.title = title
        this.position = position
        this.total = total
    }

    constructor(groupNum: Long?, title: String?, position: Long?, total: Long?) {
        this.groupNum = groupNum?.toInt() ?: 0
        this.title = title
        this.position = position?.toInt() ?: 0
        this.total = total?.toInt() ?: 0
    }

    open fun isFirst(): Boolean {
        return position == 0
    }

    open fun isLast(): Boolean {
        return position == total - 1
    }

    open fun getGroupNum(): Int {
        return groupNum
    }

    open fun setGroupNum(groupNum: Int) {
        this.groupNum = groupNum
    }

    open fun getTitle(): String? {
        return title
    }

    open fun setTitle(title: String?) {
        this.title = title
    }

    open fun getPosition(): Int {
        return position
    }

    open fun setPosition(position: Int) {
        this.position = position
    }

    open fun getTotal(): Int {
        return total
    }

    open fun setTotal(total: Int) {
        this.total = total
    }

    companion object {
        /**
         * 按26个英文字母分组
         * 存储26个组的总数
         * 多的一个备用
         */
        var totals: IntArray = IntArray(27)
        fun initTotals() {
            for (i in totals.indices) {
                totals[i] = 0
            }
        }
    }
}